src/analysis/history.hpp

```diff
@@ -17,12 +17,14 @@
 #include <string>
 #include <vector>
 
+#include "geometry/geometry.hpp"
+#include "geometry/geometry_utils.hpp"
+#include "phoebus_utils/variables.hpp"
 #include <kokkos_abstraction.hpp>
+#include <parthenon/driver.hpp>
```

Comment:
Some of these new includes are causing one of the unit tests to fail (for a stupid technical reason). Two ways to avoid: 
1. fix the build of the unit test
2. move your new reduction implementation to the `history.cpp` file and remove these includes.

I suggest 2 is easier.

Reply:
Yeah I know, it is the `geometry.hpp` flag. Why though?

Reply:
`compile_constants.hpp` is generated automatically by the cmake. The unit tests don't automatically use or rely on it. This could be fixed but would require some build system work.